src/DotVVM.Framework/ViewModel/Serialization/DefaultViewModelSerializer.cs

```diff
@@ -133,25 +135,32 @@ public void AddNewResources(IDotvvmRequestContext context)
                 context.ViewModelJson["resources"] = resourcesObject;
         }
 
-        public string BuildStaticCommandResponse(IDotvvmRequestContext context, object result)
+        public string BuildStaticCommandResponse(IDotvvmRequestContext context)
         {
             var serializer = CreateJsonSerializer();
             var viewModelConverter = new ViewModelJsonConverter(context.IsPostBack, viewModelMapper, context.Services) {
                 UsedSerializationMaps = new HashSet<ViewModelSerializationMap>()
             };
             serializer.Converters.Add(viewModelConverter);
-            var writer = new JTokenWriter();
             var response = new JObject();
+            response["result"] = WriteCommandData(context.CommandResult, serializer, "result");
+            if (context.CustomData != null) response["customData"] = WriteCommandData(context.CustomData, serializer, "custom data");
+            return response.ToString(JsonFormatting);
+        }
+
+        private static JToken WriteCommandData(object data, JsonSerializer serializer, string description)
+        {
+            var writer = new JTokenWriter();
             try
             {
-                serializer.Serialize(writer, result);
+                serializer.Serialize(writer, data);
             }
             catch (Exception ex)
             {
-                throw new Exception($"Could not serialize viewModel of type { context.ViewModel.GetType().Name }. Serialization failed at property { writer.Path }. {GeneralViewModelRecommendations}", ex);
+                throw new Exception($"Could not serialize static command {description} of type '{ data.GetType().FullName}'. Serialization failed at property { writer.Path }. {GeneralViewModelRecommendations}", ex);
```

Comment:
It's imho missing an article. "Could not serialize a static command {description} object of type ..." sounds a bit better to me.